lib/internal/source_map/source_map_cache.js

```diff
@@ -165,14 +165,16 @@ function sourceMapFromDataUrl(basePath, url) {
 // If the sources are not absolute URLs after prepending of the "sourceRoot",
 // the sources are resolved relative to the SourceMap (like resolving script
 // src in a html document).
+// TODO(bcoe): refactor to use new URL(url, base), rather than path.resolve.
 function sourcesToAbsolute(base, data) {
   data.sources = data.sources.map((source) => {
     source = (data.sourceRoot || '') + source;
+    // An absolute URI does not need to be resolved, return it:
+    if (/(^\w+):/.test(source)) return source;
```

Comment:
There might be a little edgecase with absolute windows paths which also start like that. Technically sources are urls, but not sure if everybody does that correctly.

Reply:
I switched to your original recommendation of:

```js
  data.sources = data.sources.map((source) => {
    source = (data.sourceRoot || '') + source;
    return new URL(source, baseURL).href;
  });
```

As it seemed like avoiding the refactor was creating more work. I noticed that:

`webpack:///./webpack.js` resolves as `webpack:///webpack.js`, is this any concern?

```
webpack:///webpack.js:14
  throw new Error('oh no!')
        ^

Error: oh no!
    at o (/Users/bencoe/oss/node-1/test/fixtures/source-map/webpack.js:1:970)
        -> webpack:///webpack.js:14:9
```